src/applications/auth/containers/AuthApp.jsx

```diff
@@ -113,14 +153,19 @@ export default function AuthApp({ location }) {
       requestId,
       errorCode,
     );
+    const { userProfile } = authMetrics;
+    const provisioned = await handleProvisioning(userProfile);
+    if (!provisioned) {
+      return;
+    }
     authMetrics.run();
     if (!skipToRedirect) {
-      setupProfileSession(authMetrics.userProfile);
+      setupProfileSession(userProfile);
```

Comment:
As a consequence of the above, error-free authentications with valid external redirects will now no longer pass an empty object to `AuthMetrics` and instead pass their full user profile.